sites/example-project/src/pages/query-store/+page.md

```diff
@@ -0,0 +1,9 @@
+```test_query
+SELECT * FROM marketing_spend
+```
+
+{test_query instanceof QueryStore}
+
+<DataTable data={test_query}/>
+<DataTable data={test_query.groupBy("marketing_channel").agg({ avg: ["spend"], sum: ["spend"]})}/>
```

Comment:
Good to see an example of how aggregations work!

One hand it seems pretty cool.  

On the other hand, as I interpret it, it seems like we are providing a custom Evidence DSL to get around writing aggregation directly in SQL and avoid repeating re-writing the query query ... looking at the implementation of `#withStoreCache` it seems like we are creating a new query store that simply has a new SQL query derived from the original.

So questions (only applicable if I am interpreting this right):
1. Are these aggregation JS functions intended to support a particular feature / request ? 
    * I don't think this is part of the added functionality mentioned in the description, and I am guessing this is to make it easier for the user to reuse an existing query without having to copy/paste a new query and type up a new aggregation.
3. When these queries fail at runtime, would it be easy for the user to figure out JS aggregations are bad?
   * e.g suppose the original query was  `test_query='SELECT * from marketing_spend ORDER BY spend DESC LIMIT 10'`, and they write up their <DataTable> exactly as on line 8, then I think we'd end up with a where clause after a limit, and this would be due to a bad JS aggregation.

Reply:
> looking at the implementation of #withStoreCache it seems like we are creating a new query store that simply has a new SQL query derived from the original.
This is exactly correct; the reason we do it this way (instead of mutating the query) is to ensure that you can re-use in a few places, e.g. if you paginate a query from your markdown for the datatable, we don't want it to be paginated on charts.

> I don't think this is part of the added functionality mentioned in the description, and I am guessing this is to make it easier for the user to reuse an existing query without having to copy/paste a new query and type up a new aggregation.

It is; but not for project authors

> Are these aggregation JS functions intended to support a particular feature / request ?

These are provided for _component_ development; "end users" (e.g. project authors) shouldn't have to touch or know about this API at all.

> When these queries fail at runtime, would it be easy for the user to figure out JS aggregations are bad?

The example you've point out here is a good one, I think there has to be some level of assumption that users are familiar with SQL (Especially if we format & print the query that is being run). Would be curious to hear thoughts on this

Reply:
Ok, I didn't catch this was for component dev - starting to make more sense to me!

In that case, I'd think the specificities about error is less important as it would be up the component developer to ensure the aggregations are correct. Right?